ndk/src/signers/nip46/backend/nip04-encrypt.ts

```diff
@@ -4,13 +4,29 @@ import type { IEventHandlingStrategy, NDKNip46Backend } from "./index.js";
 export default class Nip04EncryptHandlingStrategy implements IEventHandlingStrategy {
     async handle(
         backend: NDKNip46Backend,
+        id: string,
         remotePubkey: string,
         params: string[]
     ): Promise<string | undefined> {
         const [recipientPubkey, payload] = params;
         const recipientUser = new NDKUser({ hexpubkey: recipientPubkey });
-        const decryptedPayload = await backend.encrypt(remotePubkey, recipientUser, payload);
+        const decryptedPayload = await encrypt(backend, id, remotePubkey, recipientUser, payload);
 
         return decryptedPayload;
```

Comment:
here too.

Reply:
also, in the decrypt method you're returning the serialized value using `JSON.stringify` do you want to do that here too?